### ISSUE TITLE
`object-curly-newline` New option suggestion: enforce missing new line but not mention present new line

### ISSUE THREAD

**Issue by jrencz** [eslint/eslint#6771]
**What version of ESLint are you using?**
3.1.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
n/a

**What did you do? Please include the actual source code causing the issue.**
n/a

**What did you expect to happen?**
n/a

**What actually happened? Please include the actual, raw output from ESLint.**
n/a

I think it would be reasonable to allow enforcing newlines in object literals when there are at least `n` properties but allow new lines if there are less:

Suggested syntax (only for description, I don't suggest it should be like this):
```js
"object-curly-newline": ["error", {
  "minProperties": 2,
  "ifLess": "allow"
}],
```

GOOD:
```
const a = {}
const a = {foo: 1}
// This one is actually BAD with current rule capabilities
const a = {
  foo: 1
}
const a = {
  foo: 1,
  bar: 2,
}
```

BAD:
```
const a = {foo: 1,  bar: 2}
```

Rationale: 
I'd like to set the rule with `minProperties: 2` but all my object literals with one property and new lines are false negatives right now.

**Comment by mysticatea** [comment]
Thank you for this issue.

I think this is duplicate of #6488.
Please feel free to reopen if I misunderstood.

**Comment by jrencz** [comment]
@mysticatea you're right: I mean the same but I made up some different syntax to express that
